spec/requests/users_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe "User requests" do
+  describe "user registration" do
+    let!(:doorkeeper) { 
+        Doorkeeper::Application.create!(name: "Android client", redirect_uri: "", scopes: "")    
+    }
+    let!(:user) { 
+        User.create!(email: "test@example.com", password: "testxtest")    
+    }
+    let!(:token) { 
+        Doorkeeper::AccessToken.create! :application_id => doorkeeper.id, :resource_owner_id => user.id 
+    }
+
+    
+    it "Create user" do
+      post("/users", params: {email: "test@example.com", password: "testxtest", client_id: doorkeeper.uid})
+      expect(response.status).to eq(422)
+    end
+
+    it "Log in user" do
+        get("/movies", params: {Authentication: token.token, email: "test@example.com", password: "testxtest", client_id: doorkeeper.uid})
```

Comment:
It looks like you skip `doorkeeper_authorize!` for that action (movies#index) https://github.com/KMosc/APITRY/blob/059e724df1efff8fffd557c53e2143f46bb6dcfe/app/controllers/movies_controller.rb#L2
The second thing is I don't know why you are sending `token`, `email`, `password`, and `client_id` as params? I think you should read something more about how devise and doorkeeper works.

Reply:
> It looks like you skip `doorkeeper_authorize!` for that action (movies#index)
> 
> https://github.com/KMosc/APITRY/blob/059e724df1efff8fffd557c53e2143f46bb6dcfe/app/controllers/movies_controller.rb#L2
> 
> 
> The second thing is I don't know why you are sending `token`, `email`, `password`, and `client_id` as params? I think you should read something more about how devise and doorkeeper works.

If you look af previous commit, all tests were with params Authentication because I didn't use skip_authentication before for all indexes then I did exception, 2 minutes and gonna push.

Reply:
Generally, the `/movies` endpoint returns just a list of all movies without any fragile data (no personal data). IMHO It could be publicly accessed without authorization.

Reply:
> get("/movies")
It should be placed in another place. As you see other readers don't even know what you want to do,  you mix up too many things in one place.

Reply:
@w0st In last few commits I've turned on authentication and every get request does not need authentication.